pkg/client/typed/discovery/restmapper.go

```diff
@@ -0,0 +1,166 @@
+package discovery
+
+import (
+	"sync"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+)
+
+type discoveryRESTMapper struct {
+	discoveryClient DiscoveryInterface
```

Comment:
This seems like some tight coupling. I'd rather see a mapper that just needs the discovery data and not a full blown client.